File: q2486/q2486.java
class Solution {
    public int appendCharacters(String s, String t) {
        int ps = 0;
        int pt = 0;
        while (ps != s.length() && pt != t.length()) {
            if (s.charAt(ps) == t.charAt(pt)) {
                pt++;
            }
            ps++;
        }
        return t.length() - pt;
    }
}

File: q2486/q2486.py
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:
        ps = 0
        pt = 0
        while ps != len(s) and pt != len(t):
            if s[ps] == t[pt]:
                pt += 1
            ps += 1
        return len(t) - pt
